lib/Pusher.php

```diff
@@ -184,6 +184,37 @@ private function check_compatibility()
 		}
 
 	}
+
+	/**
+	 * validate number of channels and channel name format.
+	 */	
+	private function validate_channels($channels) {
+		if( count( $channels ) > 100 ) {
+			throw new PusherException('An event can be triggered on a maximum of 100 channels in a single call.');
+		}
+		
+		array_walk( $channels, array( $this, 'validate_channel' ) );
+	}
+
+	/**
+	 * Ensure a channel name is valid based on our spec
+	 */
+	private function validate_channel( $channel )
+	{
+		if ( ! preg_match( '/\A[-a-zA-Z0-9_=@,.;]+\z/', $channel ) ) {
+			throw new PusherException( 'Invalid channel name ' . $channel );
+		}
+	}
+
+	/**
+	 * Ensure a socket_id is valid based on our spec
+	 */
+	private function validate_socket_id( $socket_id )
+	{
+		if ( $socket_id && !preg_match( '/\A\d+\.\d+\z/', $socket_id ) ) {
```

Comment:
Thanks. These bugs probably come up rarely, but when they do they could be very confusing.